### PR TITLE
Improve `SignatureRequest` performance

### PR DESCRIPTION
**Pull request by FrederikBolding**
## Explanation

Alleviates some performance problems with `SignatureRequest` and related components which were repeatedly re-rendering causing at least part of a slow-down in the extension.
Fixes:
- Memoize `SignatureRequestData` and deep equal check data before re-rendering
- Memoize `getMetaMaskIdentities` used in `SignatureRequestData` component
- Memoize `getAddressBook` used in `Address` component
- Memoize `getMetadataContractName` used in `Address` component
- Memoize sanitization of payload

### CODE REVIEW

File: ui/selectors/selectors.js
Comment (danjm) [inline ui/selectors/selectors.js:829]: Nit, non-blocking: the first parameter of this second function should be `_state` and not `_tokenList`

**Review by danjm — APPROVED** [review]
LGTM, but because we want to release this today, let's get 2 more code review approvals before we merge

**Review by adonesky1 — APPROVED** [review]
This all looks right to me! Tested and is indeed faster as well!

**Review by brad-decker — APPROVED** [review]
LGTM

**Pull request merged**
